mypy/subtypes.py

```diff
@@ -214,6 +214,13 @@ def visit_callable_type(self, left: CallableType) -> bool:
                                   ignore_pos_arg_names=self.ignore_pos_arg_names)
                        for item in right.items())
         elif isinstance(right, Instance):
+            if right.type.is_protocol and right.type.protocol_members == ['__call__']:
```

Comment:
Maybe we should also support mixing protocol callables and normal callables in joins and meets?

Reply:
Hm, I will try this.